pkg/client/logs_test.go

```diff
@@ -201,3 +202,42 @@ func TestLogReaderNoError(t *testing.T) {
 		})
 	}
 }
+
+func TestLogReaderInvalidConfig(t *testing.T) {
+	testcases := []struct {
+		config           *LogConfig
+		expectedError    bool
+		expectedErrorMsg string
+	}{
+		{
+			config:           nil,
+			expectedError:    true,
+			expectedErrorMsg: "nil LogConfig provided",
+		},
+		{
+			config:           &LogConfig{},
+			expectedError:    true,
+			expectedErrorMsg: "config validation failed",
+		},
+	}
+
+	c, err := NewSonobuoyClient(nil, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range testcases {
+		_, err = c.LogReader(tc.config)
+		if !tc.expectedError && err != nil {
+			t.Errorf("Expected no error, got: %v", err)
+		}
+
+		if tc.expectedError {
+			if err == nil {
+				t.Errorf("Expected provided config to be invalid but got no error")
+			} else if !strings.Contains(err.Error(), tc.expectedErrorMsg) {
+				t.Errorf("Expected error to contain '%v', got '%v'", tc.expectedErrorMsg, err.Error())
```

Comment:
Just FYI (if you didnt know), you can also use `%q` to automatically have go quote the string.